lib/archethic/self_repair/notifier.ex

```diff
@@ -141,15 +143,15 @@ defmodule Archethic.SelfRepair.Notifier do
              ledger_operations: %LedgerOperations{transaction_movements: transaction_movements}
            }
          },
-         prev_available_nodes
+         prev_available_nodes,
+         genesis_address
        ) do
     prev_storage_nodes =
-      Election.chain_storage_nodes(address, prev_available_nodes)
+      address
+      |> Election.chain_storage_nodes(prev_available_nodes)
       |> Enum.map(& &1.first_public_key)
 
-    resolved_addresses =
-      transaction_movements
-      |> Enum.map(& &1.to)
+    resolved_addresses = [genesis_address | Enum.map(transaction_movements, & &1.to)]
```

Comment:
Here the resolved addresses for the genesis address should take in consideration the protocol version of the tx:
- During phase 1, tx will have last address as resolved movement, so the notifier should retrieve genesis address of each movement recipient and send the ShardRepair message to last address io node and genesis io node
- After phase 2, tx will have genesis address as resolved movement, so the notifier could direclty send the message to the genesis address. But legacy transaction will still have last address as resolved movement, so if the transaction is legacy, the notifier should retrieve genesis address for each movement.

And in both case we still need to put the chain genesis address of the transaction as you did.

Reply:
It was the intentation as this PR targets for now only phase1 and we don't know the protocol version of phase2.
Hence it will be simplifier once phase2 developed to adapt the notifier to include the right protocol version.